cg/apps/loqus.py

```diff
@@ -1,137 +1,108 @@
+        profile_vcf_path: Optional[Path] = None,
+        family_ped_path: Optional[Path] = None,
+        window_size: Optional[int] = None,
+        gq_threshold: Optional[int] = None,
+        hard_threshold: Optional[float] = None,
+        soft_threshold: Optional[float] = None,
     ) -> dict:
-        """Add observations from a VCF."""
-        load_call_parameters = [
-            "load",
-            "-c",
-            family_id,
-            "-f",
-            ped_path.as_posix(),
-            "--variant-file",
-            vcf_path.as_posix(),
-            "--check-profile",
-            gbcf_path.as_posix(),
-            "--hard-threshold",
-            "0.95",
-            "--soft-threshold",
-            "0.90",
-            "--gq-treshold",
-            "10",
-        ]
-        if self.analysis_type == "wgs" and vcf_sv_path:
-            load_call_parameters.extend(["--sv-variants", vcf_sv_path.as_posix()])
-
-        nr_variants = 0
-        self.process.run_command(parameters=load_call_parameters)
+        """Add observations to Loqusdb from VCF files."""
+        load_params = {
+            "--case-id": case_id,
+            "--variant-file": snv_vcf_path.as_posix(),
+            "--sv-variants": sv_vcf_path.as_posix() if sv_vcf_path else None,
+            "--check-profile": profile_vcf_path.as_posix() if profile_vcf_path else None,
+            "--family-file": family_ped_path.as_posix() if family_ped_path else None,
+            "--max-window": str(window_size) if window_size else None,
+            "--gq-treshold": str(gq_threshold) if gq_threshold else None,
+            "--hard-threshold": str(hard_threshold) if hard_threshold else None,
+            "--soft-threshold": str(soft_threshold) if soft_threshold else None,
+        }
+        load_call_params: list = ["load"] + get_flattened_dictionary(load_params)
+        self.process.run_command(parameters=load_call_params)
+        variants = 0
         for line in self.process.stderr_lines():
             line_content = line.split("INFO")[-1].strip()
             if "inserted" in line_content:
-                nr_variants = int(line_content.split(":")[-1].strip())
-
-        return dict(variants=nr_variants)
-
-    def delete_case(self, case_id: str) -> None:
-        """Remove a case from LoqusDB."""
-        delete_call_parameters = ["delete", "-c", case_id]
-
-        self.process.run_command(parameters=delete_call_parameters)
-        for line in self.process.stderr_lines():
-            if f"INFO Removing case {case_id}" in line:
-                return None
-            if f"WARNING Case {case_id} does not exist" in line:
-                raise CaseNotFoundError(f"Case {case_id} not found in loqusdb")
+                variants = int(line_content.split(":")[-1].strip())
 
-        # This should not happen. If it does, other exit messages must be handle
-        raise DeleteCaseError(f"Could not delete case {case_id}")
+        return dict(variants=variants)
```

Comment:
```suggestion
        return {variant=variant}
```
Since it is an int and not a list